build/source/engine/vegNrgFlux.f90

```diff
@@ -655,119 +581,93 @@ subroutine vegNrgFlux(&
+   if (firstFluxCall) then
     scalarLatHeatSubVapCanopy = getLatentHeatValue(canopyTempTrial)
     ! case when there is snow on the ground (EXCLUDE "snow without a layer" -- in this case, evaporate from the soil)
-    if(nSnow > 0)then
-     if(groundTempTrial > Tfreeze)then; err=20; message=trim(message)//'do not expect ground temperature > 0 when snow is on the ground'; return; end if
+    if (nSnow > 0) then
+     if (groundTempTrial > Tfreeze) then; err=20; message=trim(message)//'do not expect ground temperature > 0 when snow is on the ground'; return; end if
      scalarLatHeatSubVapGround = LH_sub  ! sublimation from snow
      scalarGroundSnowFraction  = 1._rkind
     ! case when the ground is snow-free
     else
      scalarLatHeatSubVapGround = LH_vap  ! evaporation of water in the soil pores: this occurs even if frozen because of super-cooled water
      scalarGroundSnowFraction  = 0._rkind
-    end if  ! (if there is snow on the ground)
-   end if  ! (if the first flux call)
-   !write(*,'(a,1x,10(f30.10,1x))') 'groundTempTrial, scalarLatHeatSubVapGround = ', groundTempTrial, scalarLatHeatSubVapGround
-
+    end if  ! end if there is snow on the ground
+   end if  ! end if the first flux call
    ! compute the roughness length of the ground (ground below the canopy or non-vegetated surface)
    z0Ground = z0soil*(1._rkind - scalarGroundSnowFraction) + z0Snow*scalarGroundSnowFraction     ! roughness length (m)
-
    ! compute the total vegetation area index (leaf plus stem)
    VAI        = scalarLAI + scalarSAI  ! vegetation area index
    exposedVAI = scalarExposedLAI + scalarExposedSAI  !  exposed vegetation area index
-
-   ! compute emissivity of the canopy (-)
-   if(computeVegFlux)then
+   if (computeVegFlux) then ! compute emissivity of the canopy (-)
     select case(ix_canopyEmis)
-     ! *** simple exponential function
-     case(simplExp)
+     case(simplExp) ! *** simple exponential function
       scalarCanopyEmissivity = 1._rkind - exp(-exposedVAI)                                     ! effective emissivity of the canopy (-)
-     ! *** canopy emissivity parameterized as a function of diffuse transmissivity
-     case(difTrans)
-      ! compute the exponential integral
-      scaleLAI = 0.5_rkind*exposedVAI
-      expi     = expInt(scaleLAI)
-      ! compute diffuse transmissivity (-)
-      diffuseTrans = (1._rkind - scaleLAI)*exp(-scaleLAI) + (scaleLAI**2._rkind)*expi
-      ! compute the canopy emissivity
-      scalarCanopyEmissivity = (1._rkind - diffuseTrans)*vegEmissivity
-     ! *** check we found the correct option
-     case default
+     case(difTrans) ! *** canopy emissivity parameterized as a function of diffuse transmissivity
+      scaleLAI = 0.5_rkind*exposedVAI  ! compute the exponential integral
+      expi     = expInt(scaleLAI)      ! compute the exponential integral
+      diffuseTrans = (1._rkind - scaleLAI)*exp(-scaleLAI) + (scaleLAI**2_i4b)*expi ! compute diffuse transmissivity (-)
```

Comment:
If we're swapping `rkind` to `i4b` in exponents to avoid roundoff errors, should we do the same for these kind of constants? `(1._rkind - scaleLAI)`

Reply:
In this case, `scaleLAI` is of type `rkind`, so `(1_i4b - scaleLAI)` and `(1._rkind - scaleLAI)` give identical values due to Fortran's mixed mode rules (the `1_i4b` literal would get promoted to `rkind` type).

Hypothetically, if `scaleLAI` were of type `i4b`, we may gain accuracy by using `(1_i4b - scaleLAI)` instead.